docs/plugin-write.md

```diff
@@ -0,0 +1,242 @@
+---
+id: plugin-write
+title: Writing Plugins
+---
+
+## Description
+
+Lets take a closer look at the minimal plugin instance we created in the [intro](plugin.md) part:
+
+```golang
+// Plugin is plugin instance
+type Plugin struct{}
+
+// Enable implements plugin.Plugin
+// Invoked just after initialization if plugin is already enabled.
+// Invoked every time the plugin is switched to enabled.
+func (c *Plugin) Enable() error {
+	return nil
+}
+
+// Disable implements plugin.Plugin
+// Invoked every time the plugin is switched to disabled.
+func (c *Plugin) Disable() error {
+	return nil
+}
+```
+
+Now only the base [`Plugin`](https://godoc.org/github.com/gotify/plugin-api#Plugin) interface has been implemented.
+To give it more functionalities, more interfaces can be implemented.
+
+## APIs
+
+### Intro
+
+The APIs are provided as interfaces and have generally two forms:
+
+The first form is called during plugin initialization.
+The methods are in the form of `^Register(.+)Handler$`, which plugins will implement to get a callback to the gotify engine to realize some functionalities.
```

Comment:
Only one API with this pattern exist. There are also SetXXX methods for the callbacks.